utils/_context/_scenarios/otel_collector.py

```diff
@@ -11,24 +13,46 @@
 
 class OtelCollectorScenario(DockerScenario):
     def __init__(self, name: str):
+        use_proxy = os.environ.get("OTELCOLLECTOR_PROXY", "true").lower() == "true"
         super().__init__(
             name,
             github_workflow="endtoend",
             doc="TODO",
             scenario_groups=[scenario_groups.end_to_end],
             include_postgres_db=True,
-            use_proxy=True,
+            use_proxy=use_proxy,
+            mocked_backend=use_proxy,  # Link mocked backend to proxy usage
         )
         self.library = ComponentVersion("otel_collector", "0.0.0")
 
+        collector_env: dict[str, str | None] = {
```

Comment:
This part must be moved inside `OpenTelemetryCollectorContainer` constructor. 

you can also move the `_require_api_key` from `EndToEndScenario` to `DockerScenario` to add a safe guard